server/engagement/views.py

```diff
@@ -1,51 +1,146 @@
+    date = datetime.strptime(x, "%Y-%m-%dT%H:%M:%S.%fZ")
 
-logger = logging.getLogger(__name__)
+    if timezone.is_aware(date):
+        return date
+    return timezone.make_aware(date)
 
+class CreateAttendanceSession(APIView):
+    permission_classes = [IsAdmin|IsApiKey]
 
-class InjestMessageEventView(generics.CreateAPIView):
-    permission_classes = [IsApiKey]
+    def post(self, request):
+        required_fields = ['title', 'key', 'expires']
+        for field in required_fields:
+            if field not in request.data:
+                return Response(
+                    {'error': f'{field} is required'},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
 
-    @transaction.atomic
-    def post(self, request, *args, **kwargs):
-        discord_id = request.data.get("discord_id")
-        channel_id = request.data.get("channel_id")
         try:
-            user = get_object_or_404(User, discord_id=discord_id)
-
-            (
-                stats,
-                created,
-            ) = DiscordMessageStats.objects.select_for_update().get_or_create(
-                member_id=user.id, channel_id=channel_id, defaults={"message_count": 1}
+            expires = request.data['expires']
+            try:
+                expires = parse_date(expires)
+                expires = expires.astimezone(timezone.utc)
+            except ValueError:
+                return Response(
+                    {"detail": "Invalid time format. Please use ISO format (with Z)."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                    )
+            
+            session = AttendanceSession.objects.create(
+                title=request.data['title'],
+                key=request.data['key'],
+                expires=expires
             )
 
-            if not created:
-                stats.message_count = F("message_count") + 1
-                stats.save()
-                stats.refresh_from_db()
+            return Response({
+                'session': {
+                    'session_id': session.session_id,
+                    'title': session.title,
+                    'key': session.key,
+                    'expires': session.expires
+                }
+            }, status=status.HTTP_201_CREATED)
 
-            logger.info(
-                "member %s has %d messages in channel %s",
-                user.id,
-                stats.message_count,
-                channel_id,
+        except ValueError as e:
+            return Response(
+                {'error': 'Invalid date or timestamp format'},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        except Exception as e:
+            return Response(
+                {'error': str(e)},
+                status=status.HTTP_400_BAD_REQUEST
             )
 
-            return Response(status=status.HTTP_202_ACCEPTED)
+class GetAttendanceSessions(generics.ListAPIView):
+    permission_classes = [IsAdmin|IsApiKey]
```

Comment:
I think we should remove the `IsApiKey` permission, since anyone with access to the discord bot would be able to retrieve information about the attendance sessions when in reality that's something we want to limit to admins only.